sktime/classification/distance_based/_shape_dtw.py

```diff
@@ -0,0 +1,477 @@
+                                               descriptor.
+                                               If a value is not given,
+                                               this value is tuned
+                                               by 10-fold cross-validation
+                                               on the training data.
+
+
+        shape_descriptor_functions  : string list, only applicable when the
+                                      shape_descriptor_function is
+                                      set to 'compound'.
+                                      Use a list of shape descriptor
+                                      functions at the same time.
+                                      (default = ['raw','derivative'])
+
+        metric_params               : dictionary for metric parameters
+                                      (default = None).
+        """
+        self.n_neighbors = n_neighbors
+        self.subsequence_length = subsequence_length
+        self.shape_descriptor_function = shape_descriptor_function
+        self.shape_descriptor_functions = shape_descriptor_functions
+        self.metric_params = metric_params
+        super(ShapeDTW, self).__init__()
+
+    def fit(self, X, y):
+        """
+        Method to perform training on the classifier.
+
+        Parameters
+        ----------
+        X - pandas dataframe of training data of shape [num_insts,1].
+        y - list of class labels of shape [num_insts].
+
+        Returns
+        -------
+        self : the shapeDTW object
+        """
+        # Perform preprocessing on params.
+        if not (isinstance(self.shape_descriptor_function, str)):
+            raise TypeError("shape_descriptor_function must be an 'str'. \
+                            Found '" +
+                            type(self.shape_descriptor_function).__name__ +
+                            "' instead.")
+
+        X, y = check_X_y(X, y, enforce_univariate=False)
+
+        if self.metric_params is None:
+            self.metric_params = {}
+
+        # If the shape descriptor is 'compound',
+        # calculate the appropriate weighting_factor
+        if self.shape_descriptor_function == "compound":
+            self._calculate_weighting_factor_value(X, y)
+
+        self.trainData = X
+        self.trainDataClasses = y
+
+        # get the number of attributes and instances
+        num_atts = self.trainData.iloc[0, 0].shape[0]
+        num_insts = self.trainData.shape[0]
+
+        # If a subsequence length is not given,
+        # then set it to sqrt(num_atts)
+        if self.subsequence_length is None:
+            self.subsequence_length = math.floor(math.sqrt(num_atts))
+
+        # Convert training data into a list of subsequences
+        st = SlidingWindowSegmenter(self.subsequence_length)
+        st.fit(self.trainData)
+        self.sequences = st.transform(self.trainData)
+
+        self.trainData = self.sequences
```

Comment:
is it necessary to store the training data in `self.trainData` and `self.trainDataClasses`, since we're overwriting it here anyway, and since the KNN classifier will store it too I'm assuming?

Reply:
Yeah kNN would store it yeah so that's making ShapeDTW less memory efficient so i'll get rid of that :)